lib/io/can.cpp

```diff
@@ -99,7 +99,7 @@ io::CanResult Can::listen()
   if (ioctl(socket_, FIONREAD) < sizeof(CanFrame)) { return io::CanResult::kFailure; }
   const auto data = receive();
   if (!data) { return io::CanResult::kFailure; }
-  CanFrame message                 = data.value();
+  const CanFrame message           = data.value();
```

Comment:
sorry, i know i've commented on this line before

common practice with `std::optional`s is to dereference it when required. 

so you would do something like `const auto message = receive(); if (!message) { /* ... */  }` 

and then `processors_.find(message->can_id);`

there is a bit of a style question here that i'm not sure how to resolve: why am I fine with using the pointer notation (`*x` and `x->member`) here but not for collections (`x.at(i)` vs `x[i]`)? i genuinely don't know. the honest answer is that this is how i was taught to do it. a possible justification is this: accessing the value inside an optional is O(1) just like a pointer derefrence. the same doesn't hold for collections in general. i suppose by that logic we should use the `x[i]` notation for vectors but not for maps?

Reply:
is dereferencing it as `*message` to pass into process message ok? it feels wrong, but `&message` doesn't work.
or is `.value()` the best way?

Reply:
`*message` is the correct way to do it.

`&message` doesn't make sense because it gives you a reference.